storage/src/kv.rs

```diff
@@ -1,4 +1,4 @@
-//! Traits for interacting with a storage system.
+//! Traits for interacting with a key/value store.
```

Comment:
This used to be the commonware-storage/store module, which I consistently confused with commonware-storage/qmdb/store module. Seeing as these traits are actually quite specific to key/value stores, I renamed it accordingly in hopes of reducing confusion.